src/components/ContextMenu.tsx

```diff
@@ -0,0 +1,77 @@
+/*!
+ * © 2024 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement, useMemo, useState } from 'react'
+import styled from 'styled-components'
+
+import { IconButton, IconButtonGroup } from './Button'
+import { AddComment, EditIcon } from './icons'
+import { Tooltip } from './Tooltip'
+
+export interface Actions {
+  label: string
+  action: () => void
+  icon: string
+}
+
+export interface ContextMenuProps {
+  actions: Actions[]
+}
+
+const Icons: {
+  [index: string]:
+    | React.FC<React.SVGAttributes<SVGElement>>
+    | (() => JSX.Element)
+} = {
+  AddComment: AddComment,
+  EditIcon: EditIcon,
+}
+
+const ContextMenuIconButton = styled(IconButton)`
+  &:not([disabled]):hover,
+  &:not([disabled]):focus {
+    color: #363636;
+    background-color: #f2f2f2;
+    border-color: #f2f2f2;
+  }
+`
+
+export const ContextMenu: React.FC<ContextMenuProps> = ({ actions }) => {
+  let Icon: React.FC<React.SVGAttributes<SVGElement>> | (() => JSX.Element)
+  return (
+    <>
+      <IconButtonGroup>
+        {actions.map((action) => {
+          {
+            Icon = Icons[action.icon]
```

Comment:
you can just replace const Icons on line 34 with a function that would receive a string for index and will return a rendered component. That way you don't need that variable at all.
![image](https://github.com/Atypon-OpenSource/manuscripts-style-guide/assets/40764175/d9a8b5ae-2403-4bb4-9b53-01fba4bcaf98)

But considering that you explicitly import those 2 icons and then have to create an object to address them dynamically, you might as well just do it with ternary operators inline to get the right icon as it's only 2 of them.

Reply:
OK